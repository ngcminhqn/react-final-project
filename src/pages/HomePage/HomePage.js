import { useEffect } from "react";
import Loading from "../../components/Loading";
import { useNavigate } from "react-router-dom";
import { CardStyled, HomeContainerStyled } from "./HomePage.styled";
import { Col, Row } from "antd";
import { useDispatch, useSelector } from "react-redux";
import { homeActions } from "../../features/home/homeSlice";
import QuestionItem from "./components/QuestionItem/QuestionItem";
import { PATH_NAME } from "../../constants/pathName";

const HomePage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const { loading, questions } = useSelector((state) => state.home);
  const { user } = useSelector((state) => state.auth);

  useEffect(() => {
    dispatch(homeActions.handleGetQuestions());
  }, [dispatch]);

  const handlePressQuestionItem = (questionId) => {
    navigate(PATH_NAME.POLL.replace(":questionId", questionId));
  };

  return (
    <HomeContainerStyled>
      <Loading isLoading={loading} />
      <CardStyled
        title="New Questions"
        style={{
          marginBottom: "20px",
        }}
        styles={{
          title: {
            textAlign: "center",
            fontSize: "36px",
          },
        }}
      >
        <Row gutter={[16, 16]}>
          {Object.values(questions)
            .filter((item) => !user?.questions.includes(item?.id))
            .map((question) => (
              <Col span={8} key={question?.id}>
                <QuestionItem
                  author={question?.author}
                  key={question?.id}
                  onClick={handlePressQuestionItem}
                  {...question}
                />
              </Col>
            ))}
        </Row>
      </CardStyled>
      <CardStyled
        title="Done"
        styles={{
          title: {
            textAlign: "center",
            fontSize: "36px",
          },
        }}
      >
        <Row gutter={[16, 16]}>
          {user?.questions.map((id, index) => (
            <Col span={8} key={index}>
              <QuestionItem
                key={index}
                {...questions[id]}
                onClick={handlePressQuestionItem}
              />
            </Col>
          ))}
        </Row>
      </CardStyled>
    </HomeContainerStyled>
  );
};

export default HomePage;
