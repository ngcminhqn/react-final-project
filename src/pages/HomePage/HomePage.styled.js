import { Card } from "antd";
import styled from "styled-components";

export const HomeContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
`;

export const CardStyled = styled(Card)`
  border-radius: 4px;
  width: 80%;
`;
