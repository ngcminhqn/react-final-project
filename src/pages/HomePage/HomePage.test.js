import HomePage from "./HomePage";
import { renderWithProviders } from "../../utils/test-utils";
/**
 * @jest-environment jsdom
 */
describe("Home", () => {
  it("render home", async () => {
    const { container } = renderWithProviders(<HomePage />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
