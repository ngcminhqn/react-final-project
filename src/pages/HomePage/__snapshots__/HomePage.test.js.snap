// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render home 1`] = `
<div>
  <div
    class="sc-aYaIB cSpayn"
  >
    <div
      class="ant-card ant-card-bordered sc-gEvDqW eyroAf css-dev-only-do-not-override-m4timi"
      style="margin-bottom: 20px;"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
            style="text-align: center; font-size: 36px;"
          >
            New Questions
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-row css-dev-only-do-not-override-m4timi"
          style="margin-left: -8px; margin-right: -8px; row-gap: 16px;"
        />
      </div>
    </div>
    <div
      class="ant-card ant-card-bordered sc-gEvDqW eyroAf css-dev-only-do-not-override-m4timi"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
            style="text-align: center; font-size: 36px;"
          >
            Done
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-row css-dev-only-do-not-override-m4timi"
          style="margin-left: -8px; margin-right: -8px; row-gap: 16px;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Home snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB cSpayn"
  >
    <div
      class="ant-card ant-card-bordered sc-gEvDqW eyroAf css-dev-only-do-not-override-m4timi"
      style="margin-bottom: 20px;"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
            style="text-align: center; font-size: 36px;"
          >
            New Questions
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-row css-dev-only-do-not-override-m4timi"
          style="margin-left: -8px; margin-right: -8px; row-gap: 16px;"
        />
      </div>
    </div>
    <div
      class="ant-card ant-card-bordered sc-gEvDqW eyroAf css-dev-only-do-not-override-m4timi"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
            style="text-align: center; font-size: 36px;"
          >
            Done
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="ant-row css-dev-only-do-not-override-m4timi"
          style="margin-left: -8px; margin-right: -8px; row-gap: 16px;"
        />
      </div>
    </div>
  </div>
</div>
`;
